server.py

```diff
@@ -52,12 +65,62 @@ def handle_mqtt_message(client, userdata, msg):
 
     client_name = message.topic.split("/")[1]
     if client_name in CLIENT_IDS:
-        if message == constants.DEFAULT_IMAGE_INIT:
-            initialize_new_image(client_name, dimensions, label)
-        elif message == constants.DEFAULT_IMAGE_CHUNK:
-            add_data_chunk(client_name, data)
-        elif message == constants.DEFAULT_IMAGE_END:
-            convert_data(client_name)
+        if CONFIGURATION == LearningType.FEDERATED:
+            collect_federated_data(data, message, client_name)
+        elif CONFIGURATION == LearningType.CENTRALIZED:
+            collect_centralized_data(
+                data, message, client_name, dimensions, label)
+
+
+def collect_federated_data(data, message, client_id):
+    global NETWORK
+
+    # get model
+    if message == constants.DEFAULT_NETWORK_INIT:
+        CLIENT_NETWORKS[client_id] = Networkblock()
+        CLIENT_NETWORKS[client_id].destroy_network_data()
+
+    elif message == constants.DEFAULT_NETWORK_CHUNK:
+        CLIENT_NETWORKS[client_id].add_network_chunk(data)
+
+    elif message == constants.DEFAULT_NETWORK_END:
+        state_dict = CLIENT_NETWORKS[client_id].reconstruct_state_dict()
+        person_binary_classifier = get_default_model().load_last_layer_state_dictionary(state_dict)
```

Comment:
you need to do:
```
person_binary_classifier = get_default_model()
person_binary_classifier.load_last_layer_state_dictionary(state_dict)
```
because `load_last_layer_state_dictionary` doesn't return itself